src/index.js

```diff
@@ -80,16 +78,14 @@ module.exports = function lintStaged(
       return runAll(config, shellMode, quietMode, debugMode)
         .then(() => {
           debug('linters were executed successfully!')
-          // No errors, exiting with 0
+          return Promise.resolve(0)
         })
         .catch(error => {
-          // Errors detected, printing and exiting with non-zero
-          process.exitCode = 1
-          printErrors(error)
+          printErrors(error, logger)
+          return Promise.resolve(1)
```

Comment:
I wonder if it would be cleaner to `Promise.reject()`/`Promise.resolve()` here, because the bin can handle exit codes with `.then(...).catch(...)`, and any node implementation could just use `try/catch`.

Reply:
I've changed this so that it returns a `Promise<boolean>` – whether linting passed or failed. The promise will be rejected on config errors.